src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -1344,6 +1344,44 @@ def test_aks_nodepool_add_with_ossku(self, resource_group, resource_group_locati
         # delete
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
+    
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus')
+    def test_aks_create_add_nodepool_with_motd(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        node_pool_name = self.create_random_name('c', 6)
+        node_pool_name_second = self.create_random_name('c', 6)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'node_pool_name': node_pool_name,
+            'node_pool_name_second': node_pool_name_second,
+            'ssh_key_value': self.generate_ssh_keys()
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} ' \
+                     '--nodepool-name {node_pool_name} -c 1 ' \
+                     '--ssh-key-value={ssh_key_value} ' \
+                     '--message-of-the-day="Zm9vCg=="' # foo in b64
+        self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('agentPoolProfiles[0].messageOfTheDay', 'Zm9vCg==')
+        ])
+
+        # nodepool get-upgrades
+        self.cmd('aks nodepool add '
+                 '--resource-group={resource_group} '
+                 '--cluster-name={name} '
+                 '--name={node_pool_name_second} '
+                 '--message-of-the-day="YmFyCg=="',
+                 checks=[
+                    self.check('provisioningState', 'Succeeded'),
+                    self.check('messageOfTheDay', 'YmFyCg==')
```

Comment:
this is somewhat similar to http proxy config with trusted CA, we could read the content from a file instead, as it could be large. open to suggestions.

Reply:
I find it a bit odd to let the user input an encoded string, maybe we can let the user input a normal string and then encode it in the cli?

Reply:
so the content is usually expected to be a blob of multi-line text. I think that would be more difficult to pass to CLI without encoding. I was thinking the user would either have it as-is, and invoke as e.g. `--message-of-the-day=$(cat foo.txt | base64)` or else we could do `--message-of-the-day foo.txt` and read the file as plain text + encode. 

WDYT?

Reply:
I prefer normal string then encode it in cli.

We could support both file name and raw input. This way, we first take the user's input as a filename, and if no such file exists, treat it as a normal string. I'm not sure if we need a PM to participate in the discussion though.

Reply:
for context, the real world use-case is expected to be something like quoted below. I am not sure passing it as raw input is very convenient for users. I will add file-based input and see what you think?

"
UNAUTHORIZED ACCESS TO THIS DEVICE IS PROHIBITED

You must have explicit, authorized permission to access or configure this device. Unauthorized attempts and actions to access or use this system may result in civil and/or criminal penalties. All activities performed on this device are logged and monitored.
"

Reply:
I fully support adding file-based input. But if the user provides the file name, we will convert the file content in the cli? Or does it require that the user-provided file has already been encoded?

Reply:
I was thinking the content above as plain-text in the file, and then we encode it for consumption in the api.

Reply:
something like that?